src/react/projects/spark-react/src/SprkStepper/SprkStepper.js

```diff
@@ -0,0 +1,247 @@
+      event.preventDefault();
+      this.jumpToLastTab();
+    }
+  }
+
+  retreatTab() {
+    const { activeStepIndex } = this.state;
+    const { children } = this.props;
+
+    let newIndex = activeStepIndex - 1;
+
+    if (newIndex < 0) {
+      newIndex = children.length - 1;
+    }
+
+    this.setNewActiveStep(newIndex);
+  }
+
+  advanceTab() {
+    const { activeStepIndex } = this.state;
+    const { children } = this.props;
+
+    let newIndex = activeStepIndex + 1;
+
+    if (newIndex >= children.length) {
+      newIndex = 0;
+    }
+
+    this.setNewActiveStep(newIndex);
+  }
+
+  jumpToFirstTab() {
+    this.setNewActiveStep(0);
+  }
+
+  jumpToLastTab() {
+    const { children } = this.props;
+
+    this.setNewActiveStep(children.length - 1);
+  }
+
+  setNewActiveStep(stepIndex){
+    this.setState({ activeStepIndex: stepIndex });
+  }
+
+  updateSliderPosition(position) {
+    this.setState({
+      sliderStyle: { top: position }
+    });
+  }
+
+  render() {
+    const { additionalClasses, children, idString, hasDarkBackground, ...other } = this.props;
+    const { activeStepIndex } = this.state;
+
+    let hasCarousel = false;
+    let hasDesc = false;
+    let descCount = 0;
+
+    children.forEach((child, index) => {
+      if (child.props.imgSrc) {
+        // TODO validate that if one has it they all have it
+        // hasCarousel = true;
+      }
+
+      if (child.props.children) {
+        descCount++;
+      }
+    });
+
+    if (descCount === children.length) {
+      hasDesc = true;
+    } else if (descCount > 0){
+      // true if 1 or more has a description but not all
+      console.warn("SprkStepper: If any step has a description, they all must have a description.");
+    }
+
+    // build the carousel steps as needed
+    const carousel = (
+      <SprkCarousel>
+        {children.map((childNode, index) => {
+          return (
+            <SprkCarouselStep
+              key={index} //TODO lodash or something
+              imgSrc={childNode.props.imgSrc}
+              imgAlt={childNode.props.imgAlt}
+              isSelected={activeStepIndex === index}
+            >
+
+            </SprkCarouselStep>
+          )
+        })}
+      </SprkCarousel>
+    );
+
+    let stepper = (
+      <ol
+        className={classnames(
+          'sprk-c-Stepper',
+          hasDarkBackground ? 'sprk-c-Stepper--has-dark-bg' : '',
+          additionalClasses
+        )}
+        data-sprk-stepper="container"
+        role="tablist"
+        aria-orientation="vertical" // todo change based on window width
+        data-id={idString}
+        {...other}
+      >
+        {hasDesc &&
+          <SprkStepperSlider
+            title={children[activeStepIndex].props.title} // active title
+            contents={children[activeStepIndex].props.children} // active contents
+            sliderStyle={this.state.sliderStyle}
+          />
+        }
+
+        {children.map((childNode, index) => {
+
+          let step = (
```

Comment:
instead of this being a variable that gets returned, could it just be returned?

Reply:
@bobwise  did you address this feedback with RV yet?